packages/pangea-sdk/pangea/verify_audit.py

```diff
@@ -127,22 +148,51 @@ def _verify_unpublished_membership_proof(root_hash, node_hash: str, proof: Optio
     return succeeded
 
 
-def _verify_membership_proof(tree_name: str, tree_size: int, node_hash: str, proof: Optional[str]) -> Optional[bool]:
+def _fetch_roots(tree_name: str, tree_size: int, leaf_index: Optional[int]) -> Optional[bool]:
+    global pub_roots
+
+    log_section("Fetching published roots from Arweave")
+
+    succeeded = None
+    needed_roots = {tree_size}
+    if leaf_index:
+        needed_roots |= {leaf_index, leaf_index + 1}
+    pending_roots = needed_roots - set(pub_roots.keys())
+
+    try:
+        if pending_roots:
+            pub_roots |= {int(k): v for k, v in get_arweave_published_roots(tree_name, pending_roots).items()}  # type: ignore[operator]
+            pending_roots = needed_roots - set(pub_roots.keys())
+        succeeded = True
+        if pending_roots and audit:
+            logger.debug("Published root could not be fetched from Arweave")
+            logger.debug("Fetching published roots from Pangea")
+            pub_roots |= {int(k): v for k, v in _get_roots(pending_roots).items()}  # type: ignore[operator]
```

Comment:
Here you are saving Pangea's roots to `pub_roots` without distinction, so next time that you call `_fetch_roots` these roots will not be fetched so you will be returning `succeeded` in true. I don't think that's okey, but I'm not sure if it's a problem neither due to it's a command line, but it would be nice to fix it.

Reply:
I guess yes, it's a problem because `_fetch_roots` is called iteratively to each event not just once